### ISSUE TITLE
netfilter: nftables table name silently truncated (32 byte buffer, 47 needed)

### ISSUE THREAD

**Issue by Utkal059** [checkpoint-restore/criu#3131]
### Description

`nftables_get_table()` in `criu/netfilter.c` builds the per-run nftables table name as `"inet CRIU-" + criu_run_id`, but every caller passes a 32 byte buffer while the name needs 47 bytes. The name is silently truncated and the function still reports success.

`criu_run_id` became a 36 character UUID string in commit 54795f174 ("criu: use libuuid for criu_run_id generation"):

```
"inet CRIU-"        10 chars
criu_run_id (UUID)  36 chars
NUL                  1
                  = 47 bytes required
```

All four call sites declare `char table[32]`:

- `criu/netfilter.c:161` `nftables_init_connection_lock()`
- `criu/netfilter.c:244` `nftables_lock_connection_raw()`
- `criu/net.c:3365` `nftables_lock_network_internal()`
- `criu/net.c:3472` `nftables_network_unlock()`

### Why the existing check does not catch it

```c
ret = snprintf(table, n, "inet CRIU-%s", criu_run_id);
...
if (ret < 0) {
        pr_err("Cannot generate CRIU is nftables table name\n");
        return -1;
}
return 0;
```

`snprintf()` returns the length it *would* have written, not what it wrote, so on truncation it returns 46 and the `ret < 0` test passes. The compiler cannot warn either, because the `snprintf()` sits inside `nftables_get_table()` where the destination size is only known through the `n` parameter.

### Reproducer

```c
#include <stdio.h>
#include <string.h>
#include <uuid/uuid.h>

int main(void)
{
        uuid_t uuid;
        char run_id[37], table[32];
        int ret;

        uuid_generate(uuid);
        uuid_unparse(uuid, run_id);

        ret = snprintf(table, sizeof(table), "inet CRIU-%s", run_id);

        printf("ret       = %d (>= 0, so CRIU treats this as success)\n", ret);
        printf("wanted    = inet CRIU-%s\n", run_id);
        printf("written   = %s (%zu chars)\n", table, strlen(table));
        return 0;
}
```

```
$ gcc -o repro repro.c -luuid && ./repro
ret       = 46 (>= 0, so CRIU treats this as success)
wanted    = inet CRIU-059cf9cb-3ff4-4591-a9dd-67402190e10b
written   = inet CRIU-059cf9cb-3ff4-4591-a9 (31 chars)
```

Building the same call with a visible destination makes GCC report it directly:

```
warning: output may be truncated writing up to 36 bytes into a region of size 22
note: output 47 bytes into a destination of size 32
```

### Impact

All four call sites truncate identically, so dump and restore agree on the shortened name and network locking still functions. The effect is that 15 of the 36 run id characters are discarded, weakening the unique table naming that commit 9838d34de ("criu: use unique table names for nftables based locking") introduced, and the error path that was written to catch a failed name generation cannot actually fire.

### Version

```
Version: 4.2.1
GitID: v4.2.1-433-g90dd8aea2
```

I have a fix ready and will open a PR against `criu-dev`.
